proto-sync.sh

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env bash
+
+# This script synchronyzes exonum protobuf specification files with the `exonum-proto-sources` repository, where they
+# could be used from by various Exonum clients.
+#
+# It is supposed that repo with proto files has the same branches as exonum repo. Thus, this allows to have different
+# sets of proto files for certain branch. Also, script updates proto files for current branch only. For example, in
+# order to sync files for exonum branch "feature_X", one should execute this script from the local branch "feature_X".
+#
+# This script is intended to be used by Exonum developers in order to update Exonum clients with stable changes in the
+# proto files.
+
+set -eu -o pipefail
+
+# prints a section header
+function header() {
+    local title=$1
+    local rest="========================================================================"
+    echo
+    echo "===[ ${title} ]${rest:${#title}}"
+    echo
+}
+
+if [ "$#" -ne 1 ]
+  then
+    echo "Usage:
+     $0 REV"
+    echo "Where REV is the revision of the source 'exonum' repository."
+    exit 1
+fi
+
+REV="$1"
+
+EXONUM_REPO_URI="https://github.com/exonum/exonum.git"
+EXONUM_REPO_TMP_DIR="/tmp/_exonum_repo_tmp"
+
+CURR_DIR=$(pwd)
+EXONUM_REPO_ROOT=${EXONUM_REPO_TMP_DIR}
+MAIN_PROTO_FILES_DIR=${EXONUM_REPO_ROOT}/exonum/src/proto/schema/exonum
+COMPONENTS_DIR=${EXONUM_REPO_ROOT}/components
+DST_PROTO_FILES_DIR=${CURR_DIR}/src
+CURR_BRANCH_NAME=$(git rev-parse --abbrev-ref HEAD 2>/dev/null)
+
+# Clean temporary dir from the previous iteration if any
+rm -fR ${EXONUM_REPO_TMP_DIR}
+
+header "CLONING REPO"
+
+git clone --depth 30 --single-branch --branch ${CURR_BRANCH_NAME} ${EXONUM_REPO_URI} ${EXONUM_REPO_TMP_DIR}
```

Comment:
Why `--depth 30`? Will `--depth 1` work?

Reply:
We should be able to check out some recent revision, not the latest only.

Reply:
So the script won't work if there're more than 30 commits since target revision. Can we clone the whole repo or [use the only way around](https://stackoverflow.com/a/3489576)?

Reply:
Cloning the whole repo takes 5+ times more time. 30 commits it's about a month of development for the core. BUT, your suggestion looks promising, going to try.

Reply:
The suggested approach doesn't work, you may check out the comments on SO for this solution.